client/src/features/userDataApi.js

```diff
@@ -0,0 +1,18 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+const userDataApi = createApi({
```

Comment:
get error: 
Warning: Middleware for RTK-Query API at reducerPath "api" has not been added to the store.
You must add the middleware for RTK-Query to function correctly!